File: 백준/Silver/1463. 1로 만들기/1로 만들기.cc
#include <stdio.h>
#pragma warning (disable : 4996)
#define min(x, y) (x) < (y) ? (x) : (y)
int d[1000001];
int main() {
	int n;
	scanf("%d", &n);
	d[0] = 0, d[1] = 0;
	d[2] = 1;
	for (int i = 3; i <= n; i++) {
		d[i] = 0xFFFFFFF;
	}

	
	for (int i = 1; i <= n; i++) {
		if (3 * i <= n)d[3 * i] = min(d[i] + 1, d[3 * i]);
		if (2 * i <= n)d[2 * i] = min(d[i] + 1, d[2 * i]);
		if (i + 1 <= n)d[i + 1] = min(d[i] + 1, d[i + 1]);
	}
	printf("%d", d[n]);
}

File: 백준/Silver/1463. 1로 만들기/1로 만들기.py
n = int(input())
d = [1000000]*(n+1)

d[1] = 0

for i in range(1, n+1):
    if i*3<=n:
        d[3*i] = min(d[i]+1, d[3*i])
    if i*2<=n:
        d[2*i] = min(d[i]+1, d[2*i])
    if i+1<=n:
        d[i+1] = min(d[i]+1, d[i+1])
print(d[n])